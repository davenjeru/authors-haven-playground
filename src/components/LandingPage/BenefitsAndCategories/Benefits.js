import PropTypes from 'prop-types';
import React from 'react';

const Benefits = ({ benefitsArray }) => (
  <div className="col-md-6">
    <h4 className="text-primary">Benefits And Features</h4>
    <div className="row text-left pl-5">
      {benefitsArray.map(benefit => (
        <div key={benefit.value} className="col-md-6 my-3">
          <div className="row mb-3">
            <div className="align-self-center col-10 col-md-12">
              <h5 className="text-secondary">
                {benefit.value}
                {' '}
                {benefit.emoji}
              </h5>
            </div>
          </div>
        </div>
      ))}
    </div>
  </div>
);

Benefits.propTypes = {
  benefitsArray: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    emoji: PropTypes.string,
  },
  )).isRequired,
};

export default Benefits;
