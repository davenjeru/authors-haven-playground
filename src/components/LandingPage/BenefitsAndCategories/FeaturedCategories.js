import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import React from 'react';

const FeaturedCategories = ({ featuredCategories }) => (
  <div className="col-md-6">
    <h4 className="text-primary">Featured Categories</h4>
    <ul className="list-group">
      {featuredCategories.map(category => (
        <Link key={category.name} to="#technology">
          <li className="list-group-item d-flex justify-content-between align-items-center">
            {category.name}
            <span className="badge badge-pill badge-secondary">{category.count}</span>
          </li>
        </Link>
      ))}
    </ul>
  </div>
);

FeaturedCategories.propTypes = {
  featuredCategories: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    count: PropTypes.number,
  },
  )).isRequired,
};

export default FeaturedCategories;
