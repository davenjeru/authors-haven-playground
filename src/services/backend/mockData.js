// initialize the user information that should be sent in the request body in the sign-up route
export const mockUserInfo = {
  email: 'test@email.com',
  username: 'testUsername',
  password: 'testPassword',
};

// model the response object according to our backend specification
// notice the capital M in Message
export const mockSuccessResponseObject = {
  data:
    { Message: 'Check your email for further instructions' },
};

// the backend models the errors into an object whose attributes are arrays
// the front end should extract the first error
export const mockErrorResponseObject = {
  data: {
    errors: {
      email: ['Email already exists, please login or use a different email'],
      username: ['Username already exists, please enter a different username'],
    },
  },
};
