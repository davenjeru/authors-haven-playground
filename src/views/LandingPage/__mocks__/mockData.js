import Chimammanda from '../../../assets/images/face1.jpg';
import Hos from '../../../assets/images/hos.jpg';
import Ruth from '../../../assets/images/ruth.jpg';

export const benefitsArray = [
  {
    value: 'Authentic', emoji: 'ðŸ‘Œ',
  },
  {
    value: 'Knowledgeable Authors', emoji: 'ðŸ“š',
  },
  {
    value: 'Membership', emoji: 'ðŸ’³',
  },
  {
    value: 'Affordable', emoji: 'ðŸ’¸',
  },
  {
    value: 'Amazing', emoji: 'ðŸ˜Œ',
  },
  {
    value: 'Fun', emoji: 'ðŸ™Œ',
  },
];

export const featuredCategories = [
  {
    name: 'Technology',
    count: 312,
  },
  {
    name: 'Lifestyle',
    count: 299,
  },
  {
    name: 'Science',
    count: 259,
  },
  {
    name: 'Finance',
    count: 219,
  },
  {
    name: 'Politics',
    count: 155,
  },
];

export const testimonialsArray = [
  {
    name: 'Chimammanda',
    text: 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa.',
    image: Chimammanda,
    role: 'Senior writer',
  },
  {
    name: 'Hoslack',
    text: 'Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec.',
    image: Hos,
    role: 'Reader',
  },
  {
    name: 'Ruth',
    text: 'Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Cum sociis natoque penatibus et magnis dis parturient montes.',
    image: Ruth,
    role: 'Reader',
  },
];

const mockData = {
  benefitsArray,
  featuredCategories,
  testimonialsArray,
};

export default mockData;
