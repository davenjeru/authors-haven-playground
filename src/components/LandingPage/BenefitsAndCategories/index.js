import React from 'react';
import FeaturedCategories from './FeaturedCategories';
import Benefits from './Benefits';

const BenefitsAndCategories = ({ featuredCategories, benefitsArray }) => (
  <div className="container">
    <div className="row text-center">
      <FeaturedCategories featuredCategories={featuredCategories} />
      <Benefits benefitsArray={benefitsArray} />
    </div>
  </div>
);

BenefitsAndCategories.propTypes = {
  featuredCategories: FeaturedCategories.propTypes.featuredCategories.isRequired,
  benefitsArray: Benefits.propTypes.benefitsArray.isRequired,
};

export default BenefitsAndCategories;
