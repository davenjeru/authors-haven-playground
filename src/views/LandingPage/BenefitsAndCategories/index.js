import React from 'react';
import FeaturedCategories from './FeaturedCategories';
import Benefits from './Benefits';

const BenefitsAndCategories = ({ featuredCategories, benefitsArray }) => (
  <div className="container">
    <div className="row text-center">
      <FeaturedCategories featuredCategories={featuredCategories} />
      <Benefits benefitsArray={benefitsArray} />
    </div>
  </div>
);

BenefitsAndCategories.propTypes = {
  ...FeaturedCategories.propTypes,
  ...Benefits.propTypes,
};

export default BenefitsAndCategories;
