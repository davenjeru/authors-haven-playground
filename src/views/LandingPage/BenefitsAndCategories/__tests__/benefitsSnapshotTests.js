import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import BenefitsAndCategories from '../index';
import { benefitsArray, featuredCategories } from '../../__mocks__/mockData';

describe('BenefitsAndCategories', () => {
  it('should not regress', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <BenefitsAndCategories
          benefitsArray={benefitsArray}
          featuredCategories={featuredCategories}
        />
      </MemoryRouter>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
