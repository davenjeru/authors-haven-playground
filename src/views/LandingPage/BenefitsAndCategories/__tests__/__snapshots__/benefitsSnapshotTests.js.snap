// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BenefitsAndCategories should not regress 1`] = `
<div
  className="container"
>
  <div
    className="row text-center"
  >
    <div
      className="col-md-6"
    >
      <h4
        className="text-primary"
      >
        Featured Categories
      </h4>
      <ul
        className="list-group"
      >
        <a
          href="/#technology"
          onClick={[Function]}
        >
          <li
            className="list-group-item d-flex justify-content-between align-items-center"
          >
            Technology
            <span
              className="badge badge-pill badge-secondary"
            >
              312
            </span>
          </li>
        </a>
        <a
          href="/#technology"
          onClick={[Function]}
        >
          <li
            className="list-group-item d-flex justify-content-between align-items-center"
          >
            Lifestyle
            <span
              className="badge badge-pill badge-secondary"
            >
              299
            </span>
          </li>
        </a>
        <a
          href="/#technology"
          onClick={[Function]}
        >
          <li
            className="list-group-item d-flex justify-content-between align-items-center"
          >
            Science
            <span
              className="badge badge-pill badge-secondary"
            >
              259
            </span>
          </li>
        </a>
        <a
          href="/#technology"
          onClick={[Function]}
        >
          <li
            className="list-group-item d-flex justify-content-between align-items-center"
          >
            Finance
            <span
              className="badge badge-pill badge-secondary"
            >
              219
            </span>
          </li>
        </a>
        <a
          href="/#technology"
          onClick={[Function]}
        >
          <li
            className="list-group-item d-flex justify-content-between align-items-center"
          >
            Politics
            <span
              className="badge badge-pill badge-secondary"
            >
              155
            </span>
          </li>
        </a>
      </ul>
    </div>
    <div
      className="col-md-6"
    >
      <h4
        className="text-primary"
      >
        Benefits And Features
      </h4>
      <div
        className="row text-left pl-5"
      >
        <div
          className="col-md-6 my-3"
        >
          <div
            className="row mb-3"
          >
            <div
              className="align-self-center col-10 col-md-12"
            >
              <h5
                className="text-secondary"
              >
                Authentic
                 
                ðŸ‘Œ
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-6 my-3"
        >
          <div
            className="row mb-3"
          >
            <div
              className="align-self-center col-10 col-md-12"
            >
              <h5
                className="text-secondary"
              >
                Knowledgeable Authors
                 
                ðŸ“š
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-6 my-3"
        >
          <div
            className="row mb-3"
          >
            <div
              className="align-self-center col-10 col-md-12"
            >
              <h5
                className="text-secondary"
              >
                Membership
                 
                ðŸ’³
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-6 my-3"
        >
          <div
            className="row mb-3"
          >
            <div
              className="align-self-center col-10 col-md-12"
            >
              <h5
                className="text-secondary"
              >
                Affordable
                 
                ðŸ’¸
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-6 my-3"
        >
          <div
            className="row mb-3"
          >
            <div
              className="align-self-center col-10 col-md-12"
            >
              <h5
                className="text-secondary"
              >
                Amazing
                 
                ðŸ˜Œ
              </h5>
            </div>
          </div>
        </div>
        <div
          className="col-md-6 my-3"
        >
          <div
            className="row mb-3"
          >
            <div
              className="align-self-center col-10 col-md-12"
            >
              <h5
                className="text-secondary"
              >
                Fun
                 
                ðŸ™Œ
              </h5>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
